instance.tf

```diff
@@ -0,0 +1,85 @@
+
+#creating public ec2-instance
+resource "aws_instance" "public" {
+  ami                         = "ami-002070d43b0a4f171"
+  associate_public_ip_address = true
+  instance_type               = "t3.micro"
+  key_name                    = "keypair"
+  vpc_security_group_ids      = [aws_security_group.public.id]
+  subnet_id                   = aws_subnet.public[0].id
+
+
+  tags = {
+    Name = "${var.env_code}-public"
+  }
+}
+
+#creating private ec2-instance
+resource "aws_instance" "private" {
+  ami                    = "ami-002070d43b0a4f171"
+  instance_type          = "t3.micro"
+  key_name               = "keypair"
+  vpc_security_group_ids = [aws_security_group.private.id]
+  subnet_id              = aws_subnet.private[0].id
+
+
+  tags = {
+    Name = "${var.env_code}-private"
+  }
+}
+
+
+#security for public instance
+resource "aws_security_group" "public" {
+  name        = "${var.env_code}-public"
+  description = "Allow inbound traffic"
+  vpc_id      = aws_vpc.main.id
+
+  ingress {
+    description = "TLS from public"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["100.1.162.109/32"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "${var.env_code}-public"
+  }
+}
+
+
+#security for private instance
+resource "aws_security_group" "private" {
+  name        = "${var.env_code}-private"
+  description = "Allow inbound traffic"
+  vpc_id      = aws_vpc.main.id
+
+  ingress {
+    description = "allow port 22"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["100.1.162.109/32"]
```

Comment:
I will suggest you use the vpc cidr block for the private security group. this will make sure only the resource within the same vpc can cannot to it.